monty_profiling/floppy/flop_counting/operations/trigonometry.py

```diff
@@ -0,0 +1,131 @@
+import numpy as np
+from typing import Any, Optional
+
+__all__ = [
+    "SineOperation",
+    "CosineOperation",
+    "CrossOperation",
+    "ArccosOperation",
+    "TangentOperation",
+    "ArcTangentOperation",
+    "ArcSineOperation",
+]
+
+class SineOperation:
+    """FLOP counter for sine operations."""
+
+    def count_flops(self, *args: Any, result: Any) -> Optional[int]:
+        """Count FLOPs for sine operation.
+
+        Sine is typically implemented using Taylor series or CORDIC algorithm.
+        We estimate 8 FLOPs per value based on common implementations.
+        """
+        if not self.validate_inputs(*args):
+            return None
+
+        return 8 * np.size(result)
+
+    def validate_inputs(self, *args: Any) -> bool:
+        if len(args) != 1:
+            return False
+        try:
+            np.asarray(args[0])
+            return True
+        except Exception:
+            return False
+
+
+class CosineOperation:
+    """FLOP counter for cosine operations."""
+
+    def count_flops(self, *args: Any, result: Any) -> Optional[int]:
+        """Count FLOPs for cosine operation.
+
+        Cosine is typically implemented using Taylor series or CORDIC algorithm.
+        We estimate 8 FLOPs per value based on common implementations.
```

Comment:
This is typically of the below form:

![Screenshot 2025-01-14 at 5 28 30 PM](https://github.com/user-attachments/assets/3b2c7f3b-ce7a-4f99-a15c-941a5f490217)